src/App.tsx

```diff
@@ -1,55 +1,154 @@
-import React from 'react';
+import React, { useCallback, useMemo, useState } from 'react';
+import debounce from 'lodash.debounce';
+// eslint-disable-next-line import/no-duplicates
+import IconChevronDown from '@mdi/react';
+import { mdiChevronDown, mdiClose } from '@mdi/js';
+// eslint-disable-next-line import/no-duplicates
+import IconClose from '@mdi/react';
 import './App.scss';
+import classNames from 'classnames';
 import { peopleFromServer } from './data/people';
+import { DropdownList } from './components/DropdownList';
+import { Person } from './types/Person';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [selectedPerson, setSelectedPerson]
+    = useState<Person | null>(null);
+  const [isActive, setIsActive] = useState(false);
+  const [clickCount, setClickCount] = useState(0);
+  const [query, setQuery] = useState('');
+  const [visibleQuery, setVisibleQuery] = useState('');
+
+  const toggleDropdown = () => {
+    setClickCount(clickCount + 1);
+    setIsActive(clickCount % 2 === 0);
```

Comment:
why do you need set addition state after toggle?
`const isActive = clickCount % 2 === 0`